dataland-frontend/src/components/resources/frameworkDataSearch/lksg/LksgCompanyDataTable.vue

```diff
@@ -41,28 +41,28 @@
         :key="reportingPeriod.dataId"
       >
         <template #body="{ data }">
-          <a
-            v-if="
-              Array.isArray(data[reportingPeriod.dataId]) &&
-              (data[reportingPeriod.dataId].length > 1 ||
-                data[reportingPeriod.dataId].some((el) => typeof el === 'object'))
-            "
-            @click="openModalAndDisplayValuesInSubTable(data[reportingPeriod.dataId], data.kpiLabel, data.kpiKey)"
-            class="link"
-            >Show "{{ data.kpiLabel }}"
-            <em class="material-icons" aria-hidden="true" title=""> dataset </em>
-          </a>
-          <span v-else-if="typeof data[reportingPeriod.dataId] === 'object' && data[reportingPeriod.dataId]?.value">
-            {{ data[reportingPeriod.dataId].value }}
-          </span>
-
-          <span v-else
-            >{{
-              Array.isArray(data[reportingPeriod.dataId])
-                ? data[reportingPeriod.dataId][0]
-                : data[reportingPeriod.dataId]
-            }}
-          </span>
+          <template v-if="data[reportingPeriod.dataId] !== undefined && data[reportingPeriod.dataId] !== null">
+            <template v-if="Array.isArray(data[reportingPeriod.dataId])">
+              <a
+                v-if="
+                  data[reportingPeriod.dataId].length > 1 ||
+                  data[reportingPeriod.dataId].some((el) => typeof el === 'object')
+                "
+                @click="openModalAndDisplayValuesInSubTable(data[reportingPeriod.dataId], data.kpiLabel, data.kpiKey)"
+                class="link"
+                >Show "{{ data.kpiLabel }}"
+                <em class="material-icons" aria-hidden="true" title=""> dataset </em>
+              </a>
+              <span v-else> data[reportingPeriod.dataId][0] </span>
```

Comment:
It seems that in some cases arrays are not handled and displayed correctly.   Look at this case I had locally with the fake fixture company "Achkinadze-Kreb":

![image](https://github.com/d-fine/Dataland/assets/99255296/d3ee4776-7874-4f7d-9302-c5d1f65f6093)

Reply:
Even though some locations are available, they are not displayed in the popup.

Reply:
This is a bug with how the headers are generated. I fixed it for this specific case by removing the header completely since it is kind of redundant for the single column "Countries" table. For all other cases it needs to be made sure that there is a correct specification for the headers in `DataModelsTranslations.ts`, which is rather inconvenient, I admit, so if you have a better idea we could discuss.

Reply:
Thanks, I will look into this and reply as soon as I have made up my mind about it.

Reply:
1.) Why don't we rename `tableType` to `kpiKeyOfTable` ?   At the end this just contains the kpi field name...

Reply:
2.) Why the watcher for `listOfRowContents` ?  This should not change anymore..

Reply:
3.) In `generateColsNames()`, why do you check if `listOfRowContents` is an array? It already is defined as array, so why checking that?

Reply:
4.) In `generateColumns()` we set the same value `key` in both fields of the object that we `push` into `columns`.
  So why does it even have two fields in the first place?

Reply:
7.) Typo in `humanizeStringIfNeccessary()`

Reply:
8.) This whole component only works for objects which have only one sub-level.  Shouldn't we therefore use the power of typescript and give the `listOfRowContents` a type which reflects that?

Reply:
8.) I think the component can only handle two-level objects.  Shouldn't we use the power of typescript and reflect that in the type of the prop?

e.g. `type TwoLevelObject = {
  [key: string]: {
    [subKey: string]: any;
  };
};`

Reply:
If I remember correctly, the original purpose of the DetailsCompanyDataTable component has been to display the production sites. We chose to generalize this component in order to display several subtables including list of nace codes or list of country codes or any other lists or list of objects like production sites. This may be the case why it appears complicated.

Reply:
1) sounds fine to me. I rename it

Reply:
2. I don't know - I would put it into mounted, then it will run only once in the beginning as expected

Reply:
3) & 5) : these queries are indeed deprecated. I remove them

Reply:
10.) `columnHeaders: detailsCompanyDataTableColumnHeaders`    in the `data{}` object is unnecessary. You could directly write `detailsCompanyDataTableColumnHeaders` there, and also use this in the HTML template.
This also helps developers, because if the STRG-click on it to check what this is, they get redirected to that `const` right away.

Reply:
Note: I deleted 8) because you kind of handle multi-level objects.   You display their `.value` if they have that field on the first level, otherwise you directly display the object (which will look weird if it actually happens, but since this component is generic, I think it is ok for now)

Reply:
> 4.) In `generateColumns()` we set the same value `key` in both fields of the object that we `push` into `columns`. So why does it even have two fields in the first place?

simplified it by reducing the columns from list of object with two identical strings to a list of just this string (the columns name/label/header)

Reply:
> 2.) Why the watcher for `listOfRowContents` ? This should not change anymore..

I don't know, I will put its actions (the function generateColsNames()) into mounted, then it will just execute once in the beginning as desired

Reply:
> 11.) Small renaming request. `columns` is a little misleading, since it only contains the keys of fields, whose values shall be displayed in a column. So I would suggest we rename it to smth. like `keysOfValuesForColumnDisplay` and then `v-for="keyOfValue of keysOfValuesForColumnDisplay"` in the template.

done, but with `"keyOfColumn"` instead of `"keyOfValue"` which sounds a little bit too generic to me

Reply:
> 6.) In the `mounted` block, there is already an `if`-statement which checks if the received array is an array of strings. You could directly set the `columns` array to `[{ field: this.tableType, header: this.tableType }]` there, and not execute the `generateColumns()` anymore, since it is pointless at this point. Then you wouldn't need the mapping `this.listOfRowContents = dialogRefData.listOfRowContents.map((o) => ({ [this.tableType]: o }));` anymore.

I am not sure if this works as we expect. Lets have a quick talk about it.

Reply:
> 6.) In the `mounted` block, there is already an `if`-statement which checks if the received array is an array of strings. You could directly set the `columns` array to `[{ field: this.tableType, header: this.tableType }]` there, and not execute the `generateColumns()` anymore, since it is pointless at this point. Then you wouldn't need the mapping `this.listOfRowContents = dialogRefData.listOfRowContents.map((o) => ({ [this.tableType]: o }));` anymore.

Turns out we need the mapping here, because otherwise, the contents for the case of only one column won't get displayed.

Reply:
> > 6.) In the `mounted` block, there is already an `if`-statement which checks if the received array is an array of strings. You could directly set the `columns` array to `[{ field: this.tableType, header: this.tableType }]` there, and not execute the `generateColumns()` anymore, since it is pointless at this point. Then you wouldn't need the mapping `this.listOfRowContents = dialogRefData.listOfRowContents.map((o) => ({ [this.tableType]: o }));` anymore.
> 
> Turns out we need the mapping here, because otherwise, the contents for the case of only one column won't get displayed.

Yes you are right. Sorry about that comment.   But you still can set `keysOfValuesForColumnDisplay` directly to `[ this.tableType ]` in the `if`-statement, and only execute `generateColumns()` in the `else` block.

Reply:
Perfect! Thanks for all the fixes. I am closing this now since everything has been addressed!